### PR TITLE
elfio: init at 3.8

### PR DESCRIPTION
**Pull request by invokesus**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/elfio/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  version = "3.8";
+
+  src = fetchFromGitHub {
+    owner = "serge1";
+    repo = "ELFIO";
+    rev = "Release_${version}";
+    sha256 = "1h6q4nf19wpgzrz0wycl60s200r642jv41k1ffmqncgf92qy4yyv";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/elfio/default.nix:14]:
```suggestion
```

Please never overwrite phases. This breaks many of the hooks used inside nixpkgs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/elfio/default.nix (hunk 1 above)
Comment (invokesus) [reply pkgs/development/libraries/elfio/default.nix]: Okay. Removed it. Should I add `dontBuild` or `dontConfigure` then?

Comment (SuperSandro2000) [reply pkgs/development/libraries/elfio/default.nix]: Yes, please.

File: pkgs/development/libraries/elfio/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "serge1";
+    repo = "ELFIO";
+    rev = "Release_${version}";
+    sha256 = "1h6q4nf19wpgzrz0wycl60s200r642jv41k1ffmqncgf92qy4yyv";
+  };
+
+  installPhase = ''
+      mkdir -p $out/include
+      cp -R ./elfio $out/include
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/elfio/default.nix:16]:
```suggestion
    mkdir -p $out/include
    cp -R ./elfio $out/include
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please retest if the package still works.

**Comment by bjornfor** [comment]
Another elfio expr was added to nixpkgs in the meantime: 3e231f033c5bd36ea23abaa7879c2d4a739995a4

**Pull request closed without merging**